source/features/quick-mention.tsx

```diff
@@ -110,22 +131,19 @@ async function init(signal: AbortSignal): Promise<void> {
 
 	delegate('button.rgh-quick-mention', 'click', mentionUser, {signal});
 
-	// `:first-child` avoids app badges #2630
-	// The hovercard attribute avoids `highest-rated-comment`
-	// Avatars next to review events aren't wrapped in a <div> #4844
-	// :has(fieldSelector) enables the feature only when/after the "mention" button can actually work
-	// .js-quote-selection-container selects the closest parent that contains both the new comment field and the avatar #7378
-	observe([
-		// TODO: Drop after June 2025
-		`
-		.js-quote-selection-container:has(${fieldSelector[0]})
-		:is(
-			div.TimelineItem-avatar > [data-hovercard-type="user"]:first-child,
-			a.TimelineItem-avatar
-		):not([href="/${getUsername()!}"])
-	`,
-		`[data-testid="issue-viewer-container"]:has(${fieldSelector[1]}) [class^="LayoutHelpers-module__timelineElement"] a:not([href="/${getUsername()!}"])`,
-	], add, {signal});
+	const filedSignal = new AbortController();
+	observe(fieldSelector, field => {
```

Comment:
```suggestion
	observe(fieldSelector, onetime(field => {
```
Just to ensure it's never called twice. 

---

Alternatively, you can change this pattern to something like:

```JS
const controller = new AbortController();
const field: HTMLElement | undefined = await new Promise(resolve => {
  observe(fieldSelector, resolve, AbortSignal.any(etc))
  onAbort(controller, () => resolve())
});
controller.abort()

//then continue with the standard observer
```

I think this pattern is more self contained and can be replaced easily once I finish up my element-ready replacement. 

---

But not necessary